include/cuco/detail/priority_queue.inl

```diff
@@ -0,0 +1,178 @@
+#pragma once
+#include <cmath>
+
+#include <cuco/detail/error.hpp>
+#include <cuco/detail/priority_queue_kernels.cuh>
+
+namespace cuco {
+
+template <typename T, typename Compare, bool FavorInsertionPerformance, typename Allocator>
+priority_queue<T, Compare, FavorInsertionPerformance, Allocator>::priority_queue(
+  size_t initial_capacity, Allocator const& allocator)
+  : allocator_{allocator},
+    int_allocator_{allocator},
+    t_allocator_{allocator},
+    size_t_allocator_{allocator}
+{
+  node_size_ = NodeSize;
+
+  // Round up to the nearest multiple of node size
+  int nodes = ((initial_capacity + node_size_ - 1) / node_size_);
+
+  node_capacity_      = nodes;
+  lowest_level_start_ = 1 << (int)log2(nodes);
+
+  // Allocate device variables
+
+  d_size_ = std::allocator_traits<int_allocator_type>::allocate(int_allocator_, 1);
+
+  CUCO_CUDA_TRY(cudaMemset(d_size_, 0, sizeof(int)));
+
+  d_p_buffer_size_ = std::allocator_traits<size_t_allocator_type>::allocate(size_t_allocator_, 1);
+
+  CUCO_CUDA_TRY(cudaMemset(d_p_buffer_size_, 0, sizeof(size_t)));
+
+  d_heap_ = std::allocator_traits<t_allocator_type>::allocate(
+    t_allocator_, node_capacity_ * node_size_ + node_size_);
+
+  d_locks_ =
+    std::allocator_traits<int_allocator_type>::allocate(int_allocator_, node_capacity_ + 1);
+
+  CUCO_CUDA_TRY(cudaMemset(d_locks_, 0, sizeof(int) * (node_capacity_ + 1)));
+}
+
+template <typename T, typename Compare, bool FavorInsertionPerformance, typename Allocator>
+priority_queue<T, Compare, FavorInsertionPerformance, Allocator>::~priority_queue()
+{
+  std::allocator_traits<int_allocator_type>::deallocate(int_allocator_, d_size_, 1);
+  std::allocator_traits<size_t_allocator_type>::deallocate(size_t_allocator_, d_p_buffer_size_, 1);
+  std::allocator_traits<t_allocator_type>::deallocate(
+    t_allocator_, d_heap_, node_capacity_ * node_size_ + node_size_);
+  std::allocator_traits<int_allocator_type>::deallocate(
+    int_allocator_, d_locks_, node_capacity_ + 1);
+}
+
+template <typename T, typename Compare, bool FavorInsertionPerformance, typename Allocator>
+template <typename InputIt>
+void priority_queue<T, Compare, FavorInsertionPerformance, Allocator>::push(InputIt first,
+                                                                            InputIt last,
+                                                                            cudaStream_t stream)
+{
+  const int kBlockSize = min(256, (int)node_size_);
+  const int kNumBlocks = min(64000, max(1, (int)((last - first) / node_size_)));
+
+  PushKernel<<<kNumBlocks, kBlockSize, get_shmem_size(kBlockSize), stream>>>(first,
```

Comment:
`cuco` is basically following [libcudf naming convention](https://github.com/rapidsai/cudf/blob/branch-22.08/cpp/docs/DEVELOPER_GUIDE.md#code-and-documentation-style-and-formatting).

Reply:
30cbf83, 192e263, and 9da822f